File: binding/src/network/libqnetworkcookiejar.cpp
#include <QNetworkCookie>
#include <QNetworkCookieJar>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <qnetworkcookiejar.h>
#include "libqnetworkcookiejar.h"
#include "libqnetworkcookiejar.hxx"

QNetworkCookieJar* QNetworkCookieJar_new() {
	 return new VirtualQNetworkCookieJar();
}

QNetworkCookieJar* QNetworkCookieJar_new2(QObject* parent) {
	 return new VirtualQNetworkCookieJar(parent);
}

libqt_string QNetworkCookieJar_Tr(const char* s) {
	QString _ret = QNetworkCookieJar::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_list QNetworkCookieJar_CookiesForUrl(const QNetworkCookieJar* self, const QUrl* url) {
	return self->cookiesForUrl(*url);
}

bool QNetworkCookieJar_SetCookiesFromUrl(QNetworkCookieJar* self, const libqt_list cookieList, const QUrl* url) {
	return self->setCookiesFromUrl(*cookieList, *url);
}

bool QNetworkCookieJar_InsertCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	return self->insertCookie(*cookie);
}

bool QNetworkCookieJar_UpdateCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	return self->updateCookie(*cookie);
}

bool QNetworkCookieJar_DeleteCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	return self->deleteCookie(*cookie);
}

libqt_string QNetworkCookieJar_Tr2(const char* s, const char* c) {
	QString _ret = QNetworkCookieJar::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QNetworkCookieJar_Tr3(const char* s, const char* c, int n) {
	QString _ret = QNetworkCookieJar::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
libqt_list QNetworkCookieJar_QBaseCookiesForUrl(const QNetworkCookieJar* self, const QUrl* url) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_CookiesForUrl_IsBase(true);
	return vqnetworkcookiejar->cookiesForUrl(*url);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnCookiesForUrl(const QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_CookiesForUrl_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_CookiesForUrl_Callback>(slot));
}
}

// Base class handler implementation
bool QNetworkCookieJar_QBaseSetCookiesFromUrl(QNetworkCookieJar* self, const libqt_list cookieList, const QUrl* url) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_SetCookiesFromUrl_IsBase(true);
	return vqnetworkcookiejar->setCookiesFromUrl(*cookieList, *url);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnSetCookiesFromUrl(QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_SetCookiesFromUrl_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_SetCookiesFromUrl_Callback>(slot));
}
}

// Base class handler implementation
bool QNetworkCookieJar_QBaseInsertCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_InsertCookie_IsBase(true);
	return vqnetworkcookiejar->insertCookie(*cookie);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnInsertCookie(QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_InsertCookie_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_InsertCookie_Callback>(slot));
}
}

// Base class handler implementation
bool QNetworkCookieJar_QBaseUpdateCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_UpdateCookie_IsBase(true);
	return vqnetworkcookiejar->updateCookie(*cookie);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnUpdateCookie(QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_UpdateCookie_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_UpdateCookie_Callback>(slot));
}
}

// Base class handler implementation
bool QNetworkCookieJar_QBaseDeleteCookie(QNetworkCookieJar* self, const QNetworkCookie* cookie) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_DeleteCookie_IsBase(true);
	return vqnetworkcookiejar->deleteCookie(*cookie);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnDeleteCookie(QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_DeleteCookie_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_DeleteCookie_Callback>(slot));
}
}

// Derived class handler implementation
bool QNetworkCookieJar_ValidateCookie(const QNetworkCookieJar* self, const QNetworkCookie* cookie, const QUrl* url) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
	return vqnetworkcookiejar->validateCookie(*cookie, *url);
	} else {
	return self->QNetworkCookieJar::validateCookie(*cookie, *url);
}
}

// Base class handler implementation
bool QNetworkCookieJar_QBaseValidateCookie(const QNetworkCookieJar* self, const QNetworkCookie* cookie, const QUrl* url) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_ValidateCookie_IsBase(true);
	return vqnetworkcookiejar->validateCookie(*cookie, *url);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnValidateCookie(const QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_ValidateCookie_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_ValidateCookie_Callback>(slot));
}
}

// Derived class handler implementation
libqt_list QNetworkCookieJar_AllCookies(const QNetworkCookieJar* self) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
	return vqnetworkcookiejar->allCookies();
	} else {
	return self->QNetworkCookieJar::allCookies();
}
}

// Base class handler implementation
libqt_list QNetworkCookieJar_QBaseAllCookies(const QNetworkCookieJar* self) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_AllCookies_IsBase(true);
	return vqnetworkcookiejar->allCookies();
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnAllCookies(const QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<const VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_AllCookies_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_AllCookies_Callback>(slot));
}
}

// Derived class handler implementation
void QNetworkCookieJar_SetAllCookies(QNetworkCookieJar* self, const libqt_list cookieList) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
	vqnetworkcookiejar->setAllCookies(*cookieList);
	} else {
	self->QNetworkCookieJar::setAllCookies(*cookieList);
}
}

// Base class handler implementation
void QNetworkCookieJar_QBaseSetAllCookies(QNetworkCookieJar* self, const libqt_list cookieList) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_SetAllCookies_IsBase(true);
	vqnetworkcookiejar->setAllCookies(*cookieList);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkCookieJar_OnSetAllCookies(QNetworkCookieJar* self, intptr_t slot) {
	auto* vqnetworkcookiejar = dynamic_cast<VirtualQNetworkCookieJar*>(self);
	if (vqnetworkcookiejar && vqnetworkcookiejar->isVirtualQNetworkCookieJar) {
vqnetworkcookiejar->setQNetworkCookieJar_SetAllCookies_Callback(reinterpret_cast<VirtualQNetworkCookieJar::QNetworkCookieJar_SetAllCookies_Callback>(slot));
}
}

void QNetworkCookieJar_Delete(QNetworkCookieJar* self) {
    delete self;
}

File: binding/src/network/libqnetworkcookiejar.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qnetworkcookiejar.html
pub const qnetworkcookiejar = struct {

    /// New constructs a new QNetworkCookieJar object.
    pub fn New() QtC.QNetworkCookieJar {
        return qtc.QNetworkCookieJar_new();
    }


    /// New2 constructs a new QNetworkCookieJar object.
    pub fn New2(parent: ?*anyopaque) QtC.QNetworkCookieJar {
        return qtc.QNetworkCookieJar_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkCookieJar_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkcookiejar.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn CookiesForUrl(self: ?*anyopaque, url: ?*anyopaque) []const u8 {
        return qtc.QNetworkCookieJar_CookiesForUrl(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnCookiesForUrl(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) [*:0]const u8) void {
        qtc.QNetworkCookieJar_OnCookiesForUrl(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCookiesForUrl(self: ?*anyopaque, url: ?*anyopaque) []const u8 {
        return qtc.QNetworkCookieJar_QBaseCookiesForUrl(@ptrCast(self), @ptrCast(url));
    }

    pub fn SetCookiesFromUrl(self: ?*anyopaque, cookieList: []const u8, url: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_SetCookiesFromUrl(@ptrCast(self), @ptrCast(cookieList), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnSetCookiesFromUrl(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkCookieJar_OnSetCookiesFromUrl(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetCookiesFromUrl(self: ?*anyopaque, cookieList: []const u8, url: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_QBaseSetCookiesFromUrl(@ptrCast(self), @ptrCast(cookieList), @ptrCast(url));
    }

    pub fn InsertCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_InsertCookie(@ptrCast(self), @ptrCast(cookie));
    }

    /// Allows for overriding the related default method
    pub fn OnInsertCookie(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkCookieJar_OnInsertCookie(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInsertCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_QBaseInsertCookie(@ptrCast(self), @ptrCast(cookie));
    }

    pub fn UpdateCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_UpdateCookie(@ptrCast(self), @ptrCast(cookie));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateCookie(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkCookieJar_OnUpdateCookie(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_QBaseUpdateCookie(@ptrCast(self), @ptrCast(cookie));
    }

    pub fn DeleteCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_DeleteCookie(@ptrCast(self), @ptrCast(cookie));
    }

    /// Allows for overriding the related default method
    pub fn OnDeleteCookie(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkCookieJar_OnDeleteCookie(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseDeleteCookie(self: ?*anyopaque, cookie: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_QBaseDeleteCookie(@ptrCast(self), @ptrCast(cookie));
    }

    pub fn AllCookies(self: ?*anyopaque, ) []const u8 {
        return qtc.QNetworkCookieJar_AllCookies(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnAllCookies(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) [*:0]const u8) void {
        qtc.QNetworkCookieJar_OnAllCookies(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseAllCookies(self: ?*anyopaque, ) []const u8 {
        return qtc.QNetworkCookieJar_QBaseAllCookies(@ptrCast(self));
    }

    pub fn SetAllCookies(self: ?*anyopaque, cookieList: []const u8) void {
        qtc.QNetworkCookieJar_SetAllCookies(@ptrCast(self), @ptrCast(cookieList));
    }

    /// Allows for overriding the related default method
    pub fn OnSetAllCookies(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QNetworkCookieJar_OnSetAllCookies(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetAllCookies(self: ?*anyopaque, cookieList: []const u8) void {
        qtc.QNetworkCookieJar_QBaseSetAllCookies(@ptrCast(self), @ptrCast(cookieList));
    }

    pub fn ValidateCookie(self: ?*anyopaque, cookie: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_ValidateCookie(@ptrCast(self), @ptrCast(cookie), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnValidateCookie(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkCookieJar_OnValidateCookie(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseValidateCookie(self: ?*anyopaque, cookie: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QNetworkCookieJar_QBaseValidateCookie(@ptrCast(self), @ptrCast(cookie), @ptrCast(url));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkCookieJar_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkcookiejar.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkCookieJar_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkcookiejar.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNetworkCookieJar_Delete(@ptrCast(self));
    }
};
